### ISSUE TITLE
Page number display issue

### ISSUE THREAD

**Issue by gbnewby** [gutenbergtools/ebookmaker#81]
I don't know whether this is Kobo-specific. I could not replicate this with Calibre (desktop app) because it's not showing page numbers at all, and I don't have other devices handy to check.

The issue has two aspects related to page numbers.

First, page numbers are inserted towards the right margin, but impinge on the text. In the attached image, you can see "146" towards the middle-right of the page. It interferes (overwrites) the text. This is rather distracting, and sometimes obscures an important word.
![20210506_134713](https://user-images.githubusercontent.com/23138734/117364552-e667d000-ae72-11eb-9da8-a0bda8cdf4fe.jpg)

Second, the Kobo is using these in-text page numbers for the total page count at the bottom (running footer), thus resulting in the same page # being displayed on subsequent pages. In this other image, you can see in the footer that it is page 146 of 175. With the current font size & margins, there are three page #146 (and 3-4 of every other page number). The behaviour is similar for chapter page numbers, when they are set to be visible.
![20210506_134842](https://user-images.githubusercontent.com/23138734/117364358-aa346f80-ae72-11eb-8f91-48228848e742.jpg)

I looked at the HTML source for this particular book, here: https://www.gutenberg.org/ebooks/8492 .. there is no indication that page numbers are part of the HTML markup. It seems like these numbers are either being added by ebookmaker, or the Kobo is doing something strange with the markup it gets.

Both aspects of the issue are present with any PG eBook I have looked at on the Kobo (model: Aura).

I would prefer to not have page numbers embedded in the book's text. Instead, let the device show page numbers based on the current font & margin settings. That is how other eBooks display on the Kobo -- if I have a book with 200 pages, and make the font smaller or margins larger, the footer automatically adjusts to have a larger total pagecount (perhaps 300 pages, instead of 200). The running footer, which is in the correct location (not in the margin impinging the text) adjusts automatically to the current page number. And, of course, each page number only occurs once while reading the eBook.

**Comment by eshellman** [comment]
I believe this is a device setting that you have turned on. Nothing to do with PG.

**Comment by eshellman** [comment]
just out of curiosity try this ebook to see if it's different from the pg version: https://github.com/GITenberg/The-King-in-Yellow_8492/releases/download/0.1.0/book.mobi

**Comment by gbnewby** [comment]
The GITenberg file (epub) has the same behavior.

I did partially figure out Issue 1. It appears that PG eBooks (and
GITinberg too, though they use a different generator program) display
"Adobe EPUB Page numbers." In my Kobo settings, I had that turned on. This
explains issue #1.

Here is a useful description:
https://www.mobileread.com/forums/showthread.php?t=278141

These page numbers do not come from within the unpacked .epub files
(they're not part of the HTML, and I didn't see anything in the other files
comprising the EPUBs). It looks like the device does some calculation on
where to put them.

The surprising thing is that commercial books via the Kobo store do *not*
display these "Adobe EPUB page numbers" on the right side of the screen.
I.e., I had set it to display Adobe EPUB page numbers, but the only eBooks
that actually do, that I've seen so far, are from PG. The Kobo doesn't make
those other epub files visible so I wasn't able to inspect whether there is
something in their source that quells display of the Adobe EPUB page
numbers. As mentioned in the thread above, it's a crappy display choice
that tends to overwrite words at the ends of lines.

For Issue #2, again the behavior is different with the PG eBooks: The page
numbers in the footer are based on the Adobe EPUB page numbers, so they
advance surprisingly (sometimes skipping numbers, and often staying the
same for multiple page swipes, as I previously reported). (Yes, I can turn
the footer on and off.)

With books through the commercial Kobo store, the page numbers reflect the
actual number of page swipes needed. As mentioned, this means that a book
with 200 pages in the footer might change to say there are 300 pages if I
make the font smaller and/or margins bigger. But with these Adobe EPUB page
numbers, the total number of pages stays the same. This is "incorrect" from
the reader experience: It might say the book has 200 pages, but the number
of page swipes is greater or smaller depending on font/margin/spacing
choices.

Bottom line: Is there a way to quell these Adobe EPUB page numbers? I.e.,
is there a setting in the generator that can tell devices to not base
pagination on these, but instead on what the reader is actually doing on
the device? That would make the behavior of PG eBooks match that of
commercial eBooks from the Kobo store.

Again, I don't know whether this behavior exists on other make/model
devices. Desktop viewers I have (Calibre and a Facebook epub plugin) don't
display the Adobe EPUB page numbers in the margin and don't have an option
to display running footers.

I did some quick searches and turned up many reports of these issues, but
didn't immediately find solutions. But maybe it's specific to the
generator, or even the device.

**Comment by eshellman** [comment]
maybe Kobo only does that for EPUB 2? Try the Standard Ebooks versions.

**Comment by eshellman** [comment]
I think the answer may be that your commercial "epubs" are really "kepubs". https://www.mobileread.com/forums/showthread.php?t=235524   If that's the answer, I think you can close the issue.

**Comment by gbnewby** [comment]
Yes I think your explanation is correct. It's surprising that nobody has
complained about this before, at least to the PG helpdesk inbox.

Safe to close the issue. Thanks.
